packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -340,6 +342,7 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
   Widget? _error;
 
   bool get _hasError => widget.errorText != null || widget.error != null;
+  bool get _hasHelper => widget.helperText != null || widget.helper != null;
```

Comment:
nit: put these in same order as the fields they refer to (so helper, then error)